demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/home/viewmodel/WalletViewModel.kt

```diff
@@ -104,11 +105,13 @@ class WalletViewModel(application: Application) : AndroidViewModel(application)
                     withContext(Dispatchers.IO) {
                         Mnemonics.MnemonicCode(it.seedPhrase.joinToString()).toSeed()
                     }
+                val accountIndex = getCurrentAccount().hdAccountIndex?.toInt()
+                    ?: error("The account must be set at this point")
```

Comment:
Ideally the spending key would only be derived right before it is used for spending; it would not be constructed ahead of time.

Reply:
In support of that, instead of this being an error, I would have the whole spending key be a nullable value and set to null if the hd account index is not available. But maybe that will change with multi-account support instead?

Reply:
The `spendingKey` variable is a Kotlin StateFlow with a default value of null, which changes once we start observing the stream. So yes, we could leave null instead of throwing the error. The change pushed.